freyja/read_analysis_tools.py

```diff
@@ -414,7 +422,12 @@ def process_covariants(input_bam, min_site, max_site, ref_fasta, gff_file,
         muts_final = []
         for x in [read1, read2]:
             if x is None:
-                break
+                continue
+
+            if any([val is None for val in [x.query_alignment_sequence,
+                                            x.query_alignment_qualities,
+                                            x.cigarstring]]):
+                continue
```

Comment:
Just to confirm- this handles the case of missing read attributes, likely just due to reads not mapping to the reference?

Reply:
Yeah exactly, in some cases the BAM file iterator (x) is not None, but the read attributes we need are, which were causing it to error. In #257, we address the missing cigarstring, just adding the alignment sequence and qualities for more robustness

Reply:
Great- feel free to go ahead and merge then!